datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -3898,6 +3898,7 @@ SELECT * FROM (
 ) as lhs RIGHT JOIN (
     SELECT * from right_table_no_nulls
     ORDER BY b
+    LIMIT 10
```

Comment:
Rather than modifying the existing test, can you please add a new one ? If we modify the existing test we may also inadvertently be changing the coverage of this test

Likewise for the other tests in this file that now have limits

Reply:
We can not leave the test as is because the sort in the subquery gets optimized away (rightfully), breaking the intent of the test. These kind of tests seems to be originally written with the assumption that the sort wouldn't be optimized away, and the subquery with ORDER BY was mimicking a table with an order.

Placing the limit prevents the optimizer from optimizing the sort away, keeping the intent of the test intact. Keeping the test as is gives us a degenerate test.

Once we get `WITH ORDER` support for memory tables, we will simply use a memory table with ordering instead of the subquery and those tests will be reverted back. In the interim, having the LIMIT is the safest (i.e. least test-changing) path forward